src/AssetManager/View/Helper/Asset.php

```diff
@@ -0,0 +1,114 @@
+<?php
+namespace AssetManager\View\Helper;
+
+use AssetManager\Exception\InvalidArgumentException;
+use Zend\ServiceManager\ServiceLocatorInterface;
+use Zend\View\Helper\AbstractHelper;
+use Zend\Cache\Storage\Adapter\AbstractAdapter as AbstractCacheAdapter;
+
+class Asset extends AbstractHelper
+{
+    /**
+     * @var array
+     */
+    private $config;
+
+    /**
+     * @var ServiceLocatorInterface
+     */
+    private $serviceLocator;
+
+    /**
+     * @param ServiceLocatorInterface $serviceLocator
+     */
+    public function __construct(ServiceLocatorInterface $serviceLocator)
+    {
+        $this->serviceLocator = $serviceLocator;
+        $this->config         = $serviceLocator->get('config')['asset_manager'];
+    }
+
+    /**
+     * find the file and if it exists, append its unix modification time to the filename
+     *
+     * @param string $assetsPath
+     * @param string $filename
+     * @param string $queryString
+     * @return string
+     */
+    private function elaborateFilePath($assetsPath, $filename, $queryString)
+    {
+        // resolve asset
+        $asset = $this->serviceLocator->get('AssetManager\Service\AssetManager')->getResolver()->resolve($filename);
+        if ($asset !== null) {
+
+            // append last modified date to the filepath and use a custom query string
+            return $filename . '?' . $queryString . '=' . $asset->getLastModified();
```

Comment:
Everything or only the `$queryString`? I think that is only necessary to encode the `$queryString`